### PR TITLE
Upgrade to use PCRE2

### PR DESCRIPTION
**Pull request by bazsi**
This branch changes syslog-ng to use the new PCRE2 library as the old one is being removed from distributions. It is also said to be provide performance benefits but I didn't measure those.

### CODE REVIEW

**Comment by kira-syslogng** [comment]
Build FAILURE

**Comment by bazsi** [comment]
I originally attempted to keep the pcre2_match_data structure on the stack, but my attempts were futile. Even though pcre2 seems to allow to use custom allocators, trying to cache the match_data instance in a thread local variable is not really possible, as we don't know what kind of data is being allocated with the custom allocator, _even_ if we only use the custom allocator for the match_data allocation.

Kind of sad. Tried to do a quick performance test, and I've seen a pcre based $(list-search) to do 1.5M template expansions per second, so I guess it needs to be ok.

**Comment by kira-syslogng** [comment]
Build FAILURE

**Comment by bazsi** [comment]
This should fix https://github.com/syslog-ng/syslog-ng/issues/4140

File: lib/logmatcher.c [hunk 1]
```diff
@@ -502,30 +500,28 @@ log_matcher_pcre_re_match(LogMatcher *s, LogMessage *msg, gint value_handle, con
...
   if (value_len == -1)
     value_len = strlen(value);
 
-  if (pcre_fullinfo(self->pattern, self->extra, PCRE_INFO_CAPTURECOUNT, &result.num_matches) < 0)
-    g_assert_not_reached();
-  if (result.num_matches > LOGMSG_MAX_MATCHES)
-    result.num_matches = LOGMSG_MAX_MATCHES;
-
-  gsize matches_size = 3 * (result.num_matches + 1);
-  result.matches = g_alloca(matches_size * sizeof(gint));
```
[the remark this answers is not in the document]
Comment (alltilla) [reply lib/logmatcher.c:515]: Oh sorry, I just saw that you have addressed this thought 2 weeks ago in your comment.

**Review by alltilla — APPROVED** [review]

File: lib/logmatcher.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (bazsi) [reply lib/logmatcher.c:515]: Yeah, I made an effort (too much it seems, considering the outcome) to keep that allocation on the stack or in a thread local variable.

Didn't succeed. The malloc function of the generic context is called for multiple allocations, not just for the malloc_data, but apart from the size of the allocation you don't know the reason it is being allocated..

I tracked 3 cases, and I gave up.

**Pull request merged**